composition_libs/libraries/component/source.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
```

Comment:
Agree with most of the functionality removed.    I would propose brining back the 'to_s' method since that makes displaying this really easy.

And the 'description' method.   Which passes the ObjectType from the Item.   It has been nice when displaying in tables.